pkg/clients/compute.go

```diff
@@ -25,23 +25,29 @@ import (
 	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/availabilityzones"
 	"github.com/gophercloud/gophercloud/openstack/compute/v2/flavors"
 	"github.com/gophercloud/gophercloud/openstack/compute/v2/servers"
+	"github.com/gophercloud/gophercloud/openstack/utils"
 	"github.com/gophercloud/utils/openstack/clientconfig"
 	uflavors "github.com/gophercloud/utils/openstack/compute/v2/flavors"
 
 	"sigs.k8s.io/cluster-api-provider-openstack/pkg/metrics"
 )
 
 /*
-NovaMinimumMicroversion is the minimum Nova microversion supported by CAPO
-2.60 corresponds to OpenStack Queens
+NovaSupportedVersions is the list of Nova microversion supported by CAPO
+2.60 corresponds to OpenStack Queens and 2.53 to OpenStack Pike.
 
 For the canonical description of Nova microversions, see
 https://docs.openstack.org/nova/latest/reference/api-microversion-history.html
 
-CAPO uses server tags, which were added in microversion 2.52.
+CAPO uses server tags, which were first added in microversion 2.26 and then refined
+in 2.52 so it is possible to apply them when creating a server (which is what CAPO does).
 CAPO supports multiattach volume types, which were added in microversion 2.60.
 */
-const NovaMinimumMicroversion = "2.60"
+var NovaSupportedVersions = []*utils.Version{
+	{ID: "2.1", Priority: 10, Suffix: "/v2.1/"},   // Initial microversion release, same as specifying no version
```

Comment:
I'd put the bar higher here, it's hard to correlate but we might be implicitly depending on some features only available later. Starting from Newton [^1] would make sense to me.

[^1]: https://docs.openstack.org/nova/latest/reference/api-microversion-history.html#maximum-in-newton

Reply:
Fair point! I picked 2.1 because that seems to be what is used if no version is specified, so this would have been the situation before CAPO specified a version. Basically this list then represents the history of CAPO:
- No version (=2.1)
- 2.53 (added as requirement for tags)
- 2.60 (added as requirement for multi-attach volumes)

That said, I agree that it could be a good idea to set the bar higher and 2.38 sounds like a good starting place.

Reply:
> but I'd like to make it configuration variable so someone might change it if they really want

This was also my original suggestion. I wanted to make CAPO/Gophercloud pick up the microversion from `clouds.yaml`. This way the user could set the version in there if needed and it would work the same way as for the OpenStack CLI. However, there were objections since it would be very easy to break things by setting versions that CAPO cannot work with. So now I'm trying to propose a way where CAPO still has control but it is a bit more flexible.